### ISSUE TITLE
Make Vega the default for analysis

### ISSUE THREAD

**Issue by csphang** [wri/gfw-mapbuilder#286]
- Add new Vega widgets that are a one-for-one match with existing HighCharts graphs (see Glad Alerts and Glad Alerts Vega as an example)
- **Note** we may remove the Terra-I Analysis & FORMA Alerts

### Analysis name & corresponding Vega widget (multiple languages)
- [x] Total Tree Cover Loss/Gain - `e6c01eff-0d79-4865-a431-65d6adb89589`
**GIST:** https://gist.github.com/richardbarad/3e9f2b528caa78873fa6b8364c9a588d#file-totalloss-json
- [x] Tree Cover Loss - `23c064f5-fc93-4888-9db7-6d34c8681dd6`
**GIST:** https://gist.github.com/richardbarad/3e9f2b528caa78873fa6b8364c9a588d#file-annualtcl-json
- [x] Aboveground Live Woody Biomass Density -`937a74e7-f616-4d1d-91b3-e69c68e278aa`
**GIST:** https://gist.github.com/richardbarad/3e9f2b528caa78873fa6b8364c9a588d#file-carbon_emission-json
- [x] Intact Forest Landscape - `d0d22aeb-9642-4c4d-a310-f7fb95a48c21`
**GIST:** https://gist.github.com/richardbarad/3e9f2b528caa78873fa6b8364c9a588d#file-ifl-json
- [x] Tree Cover Loss by Land Cover Class - `e2ccdd72-1611-4a41-abd6-23b5cee61fa6` **(RB Added)**
**GIST:** https://gist.github.com/richardbarad/3e9f2b528caa78873fa6b8364c9a588d#file-landcoverloss-json
- [x] GLAD Alerts (Average) - `dae27aaf-6dc6-4cc4-ac65-9901ed3f162b`
**GIST:** https://gist.github.com/richardbarad/3e9f2b528caa78873fa6b8364c9a588d#file-gladalertsmonth_av-json
- [x] GLAD Alerts - `0e2da211-a9b2-4271-a437-5de56e437df2` 
**GIST:** https://gist.github.com/richardbarad/3e9f2b528caa78873fa6b8364c9a588d#file-gladalertsmonth_av-json
- [x] GLAD Alerts (Badge) - `42784c7d-95dd-40b9-804b-9cfa94c4a4d0` **(RB Added)**
**GIST:** https://gist.github.com/richardbarad/3e9f2b528caa78873fa6b8364c9a588d#file-glad_alerts_badge-json
- [x] VIIRS Active Fires - `97b82853-d9fb-4e74-b9b3-2fac7937f38f`
**GIST:**  https://gist.github.com/richardbarad/3e9f2b528caa78873fa6b8364c9a588d#file-activefires-json
- [x] Land Cover Composition - `65dc8170-0049-48e7-94ba-fa846fcb295c`
**GIST:** https://gist.github.com/richardbarad/3e9f2b528caa78873fa6b8364c9a588d#file-landcover-json

### Analysis name & corresponding Vega widget
- [x] Total Tree Cover Loss/Gain - `a7f65bb7-c037-412f-86e1-ca75e3348892`
**GIST:** https://gist.github.com/richardbarad/5c79987d7d848af5f7da0a2d75138836#file-totalloss-json
- [x] Tree Cover Loss - `de0ee864-dccb-4cc7-8a83-532b4a7330d5`
**GIST:** https://gist.github.com/richardbarad/5c79987d7d848af5f7da0a2d75138836#file-annualloss-json
- [x] Aboveground Live Woody Biomass Density -` c844257e-dc34-42ee-b6d1-0ba7d7bc5544`
**GIST:** https://gist.github.com/richardbarad/5c79987d7d848af5f7da0a2d75138836#file-annualloss-json
- [x] Intact Forest Landscape - `3a5de64f-e1b7-41d0-8b4c-d33cdf3e2a42`
**GIST:** https://gist.github.com/richardbarad/5c79987d7d848af5f7da0a2d75138836#file-iflloss-json
- [x] Tree Cover Loss by Land Cover Class - `21b18963-393a-43d9-8351-82c4cacba1a5` **(RB Added)**
**GIST:** https://gist.github.com/richardbarad/5c79987d7d848af5f7da0a2d75138836#file-loss_lancoverclass-json
- [x] GLAD Alerts - `aca48857-e783-40b4-b67d-3f9ad99ebd76` 
**GIST:** https://gist.github.com/richardbarad/5c79987d7d848af5f7da0a2d75138836#file-glad_countmonth2-json
- [x] GLAD Alerts (Badge) - `f9e10d63-153d-4e58-81de-8b63dac2a6e6` **(RB Added)**
**GIST:** https://gist.github.com/richardbarad/5c79987d7d848af5f7da0a2d75138836#file-glad_count-json
- [x] VIIRS Active Fires - `d72db3d5-dec5-4b8b-b5bc-e0c655e0bf1f`
**GIST:**  https://gist.github.com/richardbarad/5c79987d7d848af5f7da0a2d75138836#file-activefires-json
- [x] Land Cover Composition - `f4e138a2-98f9-4f1a-9f12-a93e4e05e7c2`
**GIST:** https://gist.github.com/richardbarad/5c79987d7d848af5f7da0a2d75138836#file-landcovercomposition-json

**Comment by richardbarad** [comment]
Hi @csphang,

I have updated the analysis here module to include the new vega widgets: http://training.from-dc.com/map

I created and registered a new set of widgets which are connected to the mapbuilder datasets. I am attaching a new spreadsheet which provides an overview of the widgets IDs for these widgets. We may make some changes to the widgets, but hopefully this help you get started with this task.

I believe we may need to address task #[293](https://github.com/wri/gfw-mapbuilder/issues/293) before moving forward with this. Note that we are encountering some issues with analysis calls failing when using them with vega widgets. Thomas, is going to be looking into this now that he has started his new position, but we can still move forward with this work.

Also, the one piece which I believe might be missing here is making sure that the vega widgets support multiple languages. I am going to make a new task for this and we can chat more about how to fit it into the development timeline. 
[Mapbuilder_Layers_API_20190102.xlsx](https://github.com/wri/gfw-mapbuilder/files/2724133/Mapbuilder_Layers_API_20190102.xlsx)


Best,

Richard

**Comment by richardbarad** [comment]
Hi @csphang,

I updated the list above and also added the links to GIST. Let me know if 

As discussed, I can make new versions once we decide how to move forward with the language issue here: https://github.com/wri/gfw-mapbuilder/issues/299

**Comment by csphang** [comment]
@KaylaKremer You can use the production API for the widget ID's/configurations. For example, http://production-api.globalforestwatch.org/widget/a7f65bb7-c037-412f-86e1-ca75e3348892 (where `a7f65bb7-c037-412f-86e1-ca75e3348892` is the widget id for tree cover loss/gain)

**Comment by richardbarad** [comment]
Hi @KaylaKremer , @csphang and @SwampGuzzler.

I update the list above to include the revised widgets with the language signal added in.

**Comment by richardbarad** [comment]
Also attaching a revised analysis configuration with the new widgets added. We will probably make a few small changes to the configuration before releasing the new analysis module. 

Note that there are two versions of the GLAD analysis, one will be used for countries which have a longer time-series where we can calculate a historical average. 

[New Vegas Configuration_20190410.zip](https://github.com/wri/gfw-mapbuilder/files/3064519/New.Vegas.Configuration_20190410.zip)

**Comment by KaylaKremer** [comment]
https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/merge-of-286-and-299-branches/external.html

**Comment by richardbarad** [comment]
@KaylaKremer and @SwampGuzzler,

I tested this out and it looks like the language signal works when the analysis in run in the panel but not in the report. Can we please check to make sure that the language signal also applies to the report?

Also, what will happen for apps which include custom vegas but are not translated and thus the vegas do not include a language signal? Let's test Ethiopia to make sure it still works with this update:

http://eth.restorationatlas.org/atlas

**Comment by richardbarad** [comment]
@csphang and @KaylaKremer and @csphang 

Let's also test this update on the Georgia configuration. https://atlas.mepa.gov.ge/map?l=en

To summarize: We should test on:

1. Revised Vega base configuration

2. Georgia configuration - I update the custom analysis VEGAs to include a language signal (with only ka/en as options).

3. Ethiopia - Test to make sure VEGAs which do not have a language signal still work with this update

We should also make sure the translation using the language work both in the report and when you run the analysis directly in the layer panel.

Richard

**Comment by csphang** [comment]
Will generate a Georgia build and Ethiopia build. And add the language in the report.

@richardbarad can you expand on what you mean by the revised base configuration? Is it a different Vega configuration than what this URL uses? https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/merge-of-286-and-299-branches/external.html

**Comment by richardbarad** [comment]
@csphang. Yes, it is the revised Vega configuration at the link you shared.

**Comment by KaylaKremer** [comment]
@richardbarad @SwampGuzzler @csphang 

- Ethiopia Build: https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/ethiopia-build/external.html
- Georgia Build: https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/georgia-build/external.html

For the Ethiopia build, everything for the VEGAS works fine without specifying a language signal.

For the Georgia build, all of the reports displayed fine in the selected language expect for `Land Cover Composition`. It is missing `signals` in its widget config.

**Comment by richardbarad** [comment]
@KaylaKremer. Thanks! Looks good. For Georgia, I did not update the build to include the VEGA for land cover but will do once we release 1.4.0!

**Comment by KaylaKremer** [comment]
@richardbarad The Georgia Build should now be fixed and display the reports in their appropriate language with the exception of the `Land Cover Composition` as we discussed in yesterday's meeting!

**Comment by richardbarad** [comment]
@KaylaKremer . Tested and confirmed that it work as expected. 

Can also please test on this configuration to make sure the fix works for the gloabl analysis widgets?

https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/merge-of-286-and-299-branches/external.html

**Comment by richardbarad** [comment]
@KaylaKremer & @csphang . Are you sure that Ethiopia build is working correctly? I just tested it out and it looks like it works fine in the report but when you try to load the analysis in the layer panel the app hangs and does not show the results even in the panel even though a response is returned by the API. Please see attached screenshot.

![Vega_bug](https://user-images.githubusercontent.com/40440743/58642930-4095a900-82cc-11e9-8a09-8359a5ef2bc4.png)

**Comment by richardbarad** [comment]
@SwampGuzzler & @csphang.

Noticing that this is marked as ready for WRI testing. Please check to make sure Ethiopia works following the updates. 

I would like to test on several different configurations. The configuration we have been using was from when I first started working on VEGA so does not include all the latest updates (such as the API layers).

I am thinking we should test on the following:
- V1 latest/base configuration (including a few apps)
- Georgia Atlas
- Ethiopia Restoration Atlas
- Cameroon Forest Atlas
- Tierra Indigenous Platform 

I'll work revising these configurations to use the Vega analysis. Could we put up a .js file that we can test on which includes all the latest updates? Maybe we can test both this and the VEGA downloads at the same time provided that they have been merged into one branch?

https://github.com/wri/gfw-mapbuilder/issues/287

Best,

Richard

**Comment by richardbarad** [comment]
Attached is a revised configuration for the my GFW template on ArcGIS Online which uses the new VEGA Analysis module. 

Could you please check the configuration to make sure all looks good and once that is done put up a build using this configuration and the latest version of the code for testing? 

[index.zip](https://github.com/wri/gfw-mapbuilder/files/3249033/index.zip)

I also would like to test the other configurations noted above, but let's start with this configuration and the Ethiopia atlas configuration. 

Thanks!

**Comment by KaylaKremer** [comment]
@richardbarad Here is the build with your revised config: https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/new-gfw-template-build-test/external.html

**Comment by KaylaKremer** [comment]
@richardbarad New Ethiopia build:
https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/new-ethiopia-build/external.html

**Comment by richardbarad** [comment]
@KaylaKremer. Thanks! ET looks good.

I tested the base configuration using the url above against a few apps and noticed a few issues:

1) https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/new-gfw-template-build-test/external.html?appid=0bdf18459d5544e881a06577805f2b8f.

When you try to run an analysis in this map an error is returned in the console as shown below. It looks like the call is returning a response but the VEGA is not rendering. The default language here is Spanish, so I am not sure if this cause an issue. The VEGAs load correctly when you run a report.

![image](https://user-images.githubusercontent.com/40440743/58897156-a9fb2a80-86c5-11e9-8c37-83c5ec27eeaf.png)

2) This app is not loading and also returns an error in the console related to setting the transparency of a layer: https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/new-gfw-template-build-test/external.html?appid=9e00ce21ec00476aaf42bdbeae316fd6

It is created using a public ArcGIS Online account, and includes a graphic layer. It works correctly here when using v1.3.4: http://my.gfw-mapbuilder.org/v1.3.4/?appid=9e00ce21ec00476aaf42bdbeae316fd6

Richard

**Comment by richardbarad** [comment]
@KaylaKremer and @csphang 

I have attached the following:

1) Zipped folder with an updated base configuration for use with ArcGIS Online APPs.

2) Updated JSON Configurations for the Cameroon Forest Atlas, Georgia Forest Atlas, and Tierra Indigenous. These are included in a single zipped folder.

Can we put up build to test these configurations with the new VEGA analysis module? 
Also, do you think we can remove the Restoration Analysis Module from the base configuration?

Richard

[1.4.0.zip](https://github.com/wri/gfw-mapbuilder/files/3258058/1.4.0.zip)

[Test Configurations.zip](https://github.com/wri/gfw-mapbuilder/files/3258062/Test.Configurations.zip)

**Comment by KaylaKremer** [comment]
@richardbarad I am working on making all of the builds now. Here is the build with your new 1.4.0 config: https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/test-build-with-new-config/external.html

I will go ahead and remove all references of Restoration Analysis Module and create a separate build of that to see if it breaks anything.

**Comment by KaylaKremer** [comment]
@richardbarad Here is the test build of your new config with all the references of Restoration Analysis removed. It doesn't appear to have broken anything as far as I can tell:

https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/test-build-with-new-config-no-restoration/external.html

**Comment by KaylaKremer** [comment]
@richardbarad Tierra Indigenous Test Build: https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/test-build-with-new-config-tierra-indigenous/external.html

**Comment by KaylaKremer** [comment]
@richardbarad Georgia Test Build: https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/test-build-with-new-config-georgia/external.html

**Comment by KaylaKremer** [comment]
@richardbarad And finally, the Cameroon Build: https://alpha.blueraster.io.s3.amazonaws.com/gfw-mapbuilder/test-build-with-new-config-cameroon/external.html

**Comment by richardbarad** [comment]
@KaylaKremer. Reviewed the builds above and this is approved for release!
